src/Features/LanguageServer/CommonLanguageServerProtocol.Framework/ILifeCycleManagement.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace CommonLanguageServerProtocol.Framework;
+
+public interface ILifeCycleManager
+{
+    void Shutdown();
+
+    void Exit();
```

Comment:
concerned about these being synchronous and non-cancellable.    we should also doc the threading behavior expected here as well as ordering/concurrency (e.g. what's the difference between Shutdown/Exit?  will both always get called?  in what order?  can they be called concurrently?)

Reply:
Assuming this is mapping 1:1 with the corresponding LSP events, this is fairly well covered in the spec: https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#shutdown

Reply:
It would be *amazing* to have links to these secionts of hte spec from these methods.  the point still applies that i would expect these to at least all be async.  Cancellable is iffier for them of course.  e.g. what does it mean to 'cancel the request to shutdown'.